administrator/components/com_jedchecker/libraries/helper.php

```diff
@@ -0,0 +1,262 @@
+<?php
+/**
+ * @package    Joomla.JEDChecker
+ *
+ * @copyright  Copyright (C) 2021 Open Source Matters, Inc. All rights reserved.
+ *
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die('Restricted access');
+
+
+/**
+ * class JedcheckerHelper
+ *
+ * This is a helper class with a set of static methods used by other JEDChecker classes
+ *
+ * @since  3.0
```

Comment:
What are these for? since 3.0

This should reflect true towards the JED checker and not where its taken from.

So I would think this should say `2.4.0` as this is the next release number, due to the large amount of changes a minor increment does not seem fitting.

Reply:
We are now on `2.3.0` but I would like to push out the next release soon. So I see you fixed the https://github.com/joomla-extensions/jedchecker/issues/112 xml issue with the https://github.com/joomla-extensions/jedchecker/pull/121 pull request that makes use of this helper class.

I appreciate all the contribution, but I can only go that fast with the little support we have to test and review your reposed improvements and additions.

Therefor at the moment I would just like to get the xml issue resolved and then the current staging branch thoroughly tested by the team, with a beta release from the staging branch.

Reply:
I added this since-tag to bypass a warning from Joomla's code-style check (via phpcs). With such a lot of proposed changes I expected next release will be 3.0, but as you are going to release 2.4.0, let's update this tag appropriately.

Reply:
Done,
PS. Most likely other new files contain "since 3.0" as well, please check it before merge.

Reply:
please check the function comments also

Reply:
Far as I know we don't break backward compatibility so we don't need to make a major version increment... unless I missed something. So far all seems to have a simple update path, and nothing majorly effecting the end user.

Reply:
I would like to rather make smaller steps, so we don't have to jump from 2.3 to 2.4 like this... we have to move slower.

Reply:
but we do this first step forward and then we see... how we bring J4 changes in, little by little... I hope.

Reply:
Done ("since 2.4" for functions).

Reply:
As to backward compatibility, note that JEDChecker 2.3.0 requires PHP 5.3+, but next release will require 5.6+ because of the following line in JEDCheckerHelper
```php
    public static function cleanPhpCode($content, $options = self::CLEAN_HTML | self::CLEAN_COMMENTS)
```
(of course, it's possible to introduce and use a constant `CLEAN_HTML_AND_COMMENTS = 3`, but it makes default value not so clear).

Also I'm not sure about changes in minimal Joomla version, because I'm testing JEDChecker with J!3.9 and 4.0 only (JEDChecker's XML manifest says minimal version is 3.6, but maybe it will be necessary to check it before release).